pkg/sql/sem/tree/eval_test.go

```diff
@@ -1231,6 +1231,15 @@ func TestEval(t *testing.T) {
 		{`extract(microsecond from '2010-01-10 12:13:14.123456+00:00'::timestamp)`, `123456`},
 		{`extract(epoch from '2010-01-10 12:13:14.1+00:00'::timestamp)`, `1263125594`},
 		// Extract from intervals.
+		{`extract(hour from '123m')`, `2`},
+		{`extract(hour from '123m'::interval)`, `2`},
+		{`extract(minute from '123m10s'::interval)`, `123`},
+		{`extract(second from '10m20s30ms'::interval)`, `620`},
+		{`extract(millisecond from '20s30ms40µs'::interval)`, `20030`},
+		{`extract(microsecond from '12345ns'::interval)`, `12`},
```

Comment:
Can we also test `extract(x from interval '...')` syntax? does this parse properly?